server/models/campaign.js

```diff
@@ -7,7 +7,7 @@ const CampaignSchema = new Schema({
   ownedBy: { type: Schema.Types.ObjectId, ref: "User", index: true },
   prospects: [
     {
-      prospectId: { type: Schema.Types.ObjectId, ref: "Prospect", index: true },
+      _id: { type: Schema.Types.ObjectId, ref: "Prospect", index: true },
```

Comment:
I actually liked it better when you used prospectId, I don't think it's a good idea to use an internal naming type like _id here, since those are usually used for autogenerated keys

Also when you use this, you are going to use .populate (https://mongoosejs.com/docs/populate.html) so a name like prospect or prospectId works better

Reply:
Nice job indexing 👍

Reply:
Thanks. I did this as I thought it made it easier to identify and delete a range of subdocuments using the subdocument method `.id()`. I've now reverted and found you can treat the subdocuments like a regular array and can use .splice() to delete